File: docs/api/examples/python/render.py
import json

from argentum import Argentum

client = Argentum()  # reads ARGENTUM_API_KEY and ARGENTUM_BASE_URL

me = client.me()
print(f"key \"{me['key']['name']}\" on {me['company']['name']}, scopes: {', '.join(me['key']['scopes'])}")

with open("spec.json") as f:
    spec = json.load(f)

pdf = client.reports.render(spec)
with open("revenue-python.pdf", "wb") as f:
    f.write(pdf)

print(f"wrote revenue-python.pdf ({len(pdf)} bytes)")

File: docs/api/examples/node/render.mjs
import { readFile, writeFile } from 'node:fs/promises';
import { Argentum } from '@argentum/sdk';

const client = new Argentum(); // reads ARGENTUM_API_KEY and ARGENTUM_BASE_URL

const me = await client.me();
console.log(`key "${me.key.name}" on ${me.company.name}, scopes: ${me.key.scopes.join(', ')}`);

const spec = JSON.parse(await readFile('spec.json', 'utf8'));
const pdf = await client.reports.render(spec);
await writeFile('revenue-node.pdf', pdf);

console.log(`wrote revenue-node.pdf (${pdf.length} bytes)`);
